app/vacancy_sources/my_new_term_vacancy_source.rb

```diff
@@ -0,0 +1,163 @@
+  API_KEY = ENV.fetch("VACANCY_SOURCE_MY_NEW_TERM_API_KEY").freeze
+  SOURCE_NAME = "my_new_term".freeze
+
+  def self.source_name
+    SOURCE_NAME
+  end
+
+  def initialize
+    @api_key = API_KEY
+    authenticate
+  end
+
+  def each
+    results.each do |result|
+      v = Vacancy.find_or_initialize_by(
+        external_source: SOURCE_NAME,
+        external_reference: result["reference"],
+      )
+
+      # An external vacancy is by definition always published
+      v.status = :published
+      # Consider publish_on date to be the first time we saw this vacancy come through
+      # (i.e. today, unless it already has a publish on date set)
+      v.publish_on ||= Date.today
+
+      begin
+        v.assign_attributes(attributes_for(result))
+      rescue ArgumentError => e
+        v.errors.add(:base, e)
+      end
+
+      yield v
+    end
+  end
+
+  private
+
+  def attributes_for(item)
+    {
+      job_title: item["jobTitle"],
+      job_advert: item["jobAdvert"],
+      salary: item["salary"],
+      expires_at: Time.zone.parse(item["expiresAt"]),
+      external_advert_url: item["advertUrl"],
+      job_role: job_role(item),
+      ect_status: ect_status_for(item),
+      subjects: item["subjects"].presence,
+      working_patterns: item["workingPatterns"].presence,
+      contract_type: item["contractType"]&.first,
+      phases: phases_for(item),
+      key_stages: key_stages_for(item),
+
+      # TODO: What about central office/multiple school vacancies?
+      job_location: :at_one_school,
+    }.merge(organisation_fields(item))
+  end
+
+  def organisation_fields(item)
+    schools = find_schools(item)
+    first_school = schools.first
+
+    {
+      organisations: schools,
+      readable_job_location: first_school&.name,
+      about_school: first_school&.description,
+    }
+  end
+
+  def find_schools(item)
+    multi_academy_trust = SchoolGroup.trusts.find_by(uid: item["trustUID"])
+
+    multi_academy_trust&.schools&.where(urn: item["schoolUrns"]).presence ||
+      Organisation.where(urn: item["schoolUrns"]).presence ||
+      Array(multi_academy_trust)
+  end
+
+  def job_role(item)
+    item["jobRole"].presence
+    &.gsub("headteacher", "senior_leader")
+    &.gsub("headteacher_principal", "senior_leader")
+    &.gsub("deputy_headteacher_principal", "senior_leader")
+    &.gsub("head_of_year", "middle_leader")
+    &.gsub("assistant_headteacher_principal", "middle_leader")
+    &.gsub("deputy_headteacher_principal", "middle_leader")
+    &.gsub("learning_support", "education_support")
+    &.gsub("other_support", "education_support")
+    &.gsub("science_technician", "other_education_role")
+    &.gsub(/\s+/, "")
+  end
+
+  def ect_status_for(item)
+    return unless item["ectSuitable"].presence
+
+    item["ectSuitable"] == true ? "ect_suitable" : "ect_unsuitable"
+  end
+
+  def key_stages_for(item)
+    item["keyStages"].presence&.map do |key_stage|
```

Comment:
Question/suggestion:
Isn't the `&.` redundant after `presence` ?
```suggestion
    item["keyStages"].presence.map do |key_stage|
```